### PR TITLE
Resolve issue in filter/validate deprecation path

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
@emma58 pointed out a bug in the Set filter/validate deprecation path where Scalar sets with dim=2 would silently not call filter correctly (as the deprecation path required the filter function to raise an exception or return false when called with the new syntax. This PR resolves the error and adds a test for this scenario. 

In addition, @mfripp pointed out on SO that we were being rather rude deprecating an old syntax in the same release that we added support for the new API. This PR silences the deprecation warning for the (common) case where filter / validate was being used on a Scalar Set. 

## Changes proposed in this PR:
- fix filter/validate deprecation path for ScalarSets
- add a test for the edge case that motivated this PR
- Silence (for now) the deprecation warning for scalar sets. 

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/tests/unit/test_set.py [hunk 1]
```diff
@@ -4374,12 +4399,15 @@ def _validate(model, i, j):
...
-                r"callback to match the signature \(block, value\)",
-            )
+            # Note that we are not emitting a deprecation warning (yet)
+            # for scalar sets
+            # self.assertEqual(output.getvalue(), "")
+            #     output.getvalue().replace('\n', ' '),
+            #     r"DEPRECATED: OrderedScalarSet I2: 'validate=' callback "
+            #     r"signature matched \(block, \*value\).  Please update the "
+            #     r"callback to match the signature \(block, value\)",
+            # )
```
Comment (mrmundt) [inline pyomo/core/tests/unit/test_set.py:4409]: Is this still supposed to be here?

Comment (jsiirola) [reply pyomo/core/tests/unit/test_set.py:4409]: Sorry - this was leaving a breadcrumb (with no explanation).  Eventually we will (probably?) want to turn the deprecation warning back on, in which case this test will have to revert back.

**Review by emma58 — APPROVED** [review]
Thank you for fixing this! :)

**Review by blnicho — APPROVED** [review]
